### DISCUSSION TITLE
G7 sensor stopped/killed while pairing?

### DISCUSSION THREAD

**Discussion by kameamea** [NightscoutFoundation/xDrip#3236] [Bug reports]
xdrip version: 2023.07.16
phone: Huawei P30, Android 12

Yesterday I tried to attach a new G7 sensor.
I first changed data source from xdrip+ sync follower to G5/G6/G7/1 Transmitter.
Then I inserted the 4 digit code of the sensor
Shortly after that I was prompted to confirm the bluetooth  pairing
Immediately after this confirmation the sensor was reported to be stopped.
Any attempt to get it started again with xdrip or dexcom app failed.

The log file seems to show that xdrip stopped this sensor, but why?

Is it due to the "Sensor Raw Data Sanity Failure"?

After removing the sensor I realized that some blood had leaked out, may this be the reason?

Anyway I would have prefered to be able to get some error message from the dexcom app later on to receive a replacement by dexcom. This way the dexcom app just didn't find the sensor at all. But maybe I misunderstand the meaning of "marking sensor as stopped"?
![Screenshot_20231211_200103_com eveningoutpost dexdrip](https://github.com/NightscoutFoundation/xDrip/assets/7243765/06177c5a-f3f2-403f-923c-ed17d5055afd)

**Comment by Navid200** [comment]
That log only means that xDrip has found out that the sensor has stopped.
xDrip never stops a sensor on its own.
If you tap on stop sensor, xDrip will ask you to confirm.  So, xDrip is not trigger happy with stopping sensors.  It's actually very cautious about that.

So, you used the applicator to insert the sensor, and used xDrip to connect to it?  Is that all you did?  
Or did you also use Dexcom app?

**Comment by Navid200** [comment]
Thanks for reporting this.  I am looking at the xDrip code trying to figure out what could have happened.  
I will update as soon as I find out.

**Comment by Navid200** [comment]
I admit that log is confusing.  It confused me.  
Even though it says sensor stopped at ... and gives the current time, it is really only marking the sensor as stopped.  It is not stopping the sensor.
